<a href="https://colab.research.google.com/github/nadusupalli/nadusupalli/blob/main/Week1_Day2_Python_Basics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Week 1, Day 2: Python Basics - Data Types, Variables, and Operators

## Learning Objectives
- Understand basic Python data types
- Learn about variables and variable assignment
- Master different types of operators in Python
- Practice basic Python operations


## 1. Python Data Types

### Numeric Types
- int (Integer)
- float (Floating-point)
- complex (Complex numbers)

### Text Type
- str (String)

### Boolean Type
- bool (True/False)


In [ ]:
# Examples of different data types
integer_num = 42
float_num = 3.14
complex_num = 2 + 3j
text = "Hello, Python!"
is_python_fun = True

# Check types
print(f"Type of {integer_num}: {type(integer_num)}")
print(f"Type of {float_num}: {type(float_num)}")
print(f"Type of {complex_num}: {type(complex_num)}")
print(f"Type of {text}: {type(text)}")
print(f"Type of {is_python_fun}: {type(is_python_fun)}")

Type of 42: <class 'int'>
Type of 3.14: <class 'float'>
Type of (2+3j): <class 'complex'>
Type of Hello, Python!: <class 'str'>
Type of True: <class 'bool'>


## 2. Variables

Rules for Python variables:
- Must start with a letter or underscore
- Can contain letters, numbers, and underscores
- Case-sensitive
- Cannot use reserved keywords

In [ ]:
# Variable assignment examples
name = "Alice"
age = 25
height = 1.75
_private_var = "secret"

# Multiple assignment
x, y, z = 1, 2, 3

print(f"Name: {name}")
print(f"Age: {age}")
print(f"Height: {height}")
print(f"x: {x}, y: {y}, z: {z}")

Name: Alice
Age: 25
Height: 1.75
x: 1, y: 2, z: 3


## 3. Operators

### Arithmetic Operators
- `+` (Addition)
- `-` (Subtraction)
- `*` (Multiplication)
- `/` (Division)
- `//` (Floor Division)
- `%` (Modulus)
- `**` (Exponentiation)

In [ ]:
# Arithmetic operations
a = 10
b = 3

print(f"Addition: {a + b}")
print(f"Subtraction: {a - b}")
print(f"Multiplication: {a * b}")
print(f"Division: {a / b}")
print(f"Floor Division: {a // b}")
print(f"Modulus: {a % b}")
print(f"Exponentiation: {a ** b}")

Addition: 13
Subtraction: 7
Multiplication: 30
Division: 3.3333333333333335
Floor Division: 3
Modulus: 1
Exponentiation: 1000


## Practical Exercises

In [ ]:
# Exercise 1: Temperature Converter
def celsius_to_fahrenheit(celsius):
    return (celsius * 9/5) + 32

# Test the function
temp_c = 25
temp_f = celsius_to_fahrenheit(temp_c)
print(f"{temp_c}°C is equal to {temp_f}°F")

25°C is equal to 77.0°F


In [ ]:
# Exercise 2: Simple Interest Calculator
def calculate_simple_interest(principal, rate, time):
    return (principal * rate * time) / 100

# Test the function
p = 1000  # Principal amount
r = 5     # Rate of interest
t = 2     # Time in years

interest = calculate_simple_interest(p, r, t)
print(f"Simple Interest: ${interest}")

Simple Interest: $100.0


## MCQ Quiz

1. Which of the following is not a valid variable name in Python?
   - a) my_var
   - b) _variable
   - c) 1variable
   - d) myVariable

2. What is the output of 17 // 5?
   - a) 3.4
   - b) 3
   - c) 2
   - d) 3.0

3. Which operator is used for exponentiation in Python?
   - a) ^
   - b) **
   - c) //
   - d) %%

4. What is the type of 5.0?
   - a) int
   - b) float
   - c) double
   - d) number

5. What will be the value of x after x = 2 + 3 * 4?
   - a) 20
   - b) 14
   - c) 24
   - d) 10

6. Which of these is a valid way to assign multiple variables?
   - a) a, b, c = 1, 2
   - b) a, b, c = 1, 2, 3
   - c) a, b = 1
   - d) a = b = 1, 2, 3

7. What is the result of 10 % 3?
   - a) 3
   - b) 1
   - c) 0
   - d) 3.33

8. Which of these is not a numeric type in Python?
   - a) int
   - b) float
   - c) complex
   - d) char

9. What is the type of True in Python?
   - a) string
   - b) boolean
   - c) bool
   - d) int

10. What will be the output of type("123")?
    - a) int
    - b) float
    - c) str
    - d) number

Answers: 1-c, 2-b, 3-b, 4-b, 5-b, 6-b, 7-b, 8-d, 9-c, 10-c